numpy/lib/function_base.py

```diff
@@ -671,11 +671,23 @@ def select(condlist, choicelist, default=0):
         raise ValueError("select with an empty condition list is not possible")
 
     choicelist = [np.asarray(choice) for choice in choicelist]
-    choicelist.append(np.asarray(default))
+
+    try:
+        intermediate_dtype = np.result_type(*choicelist)
+    except TypeError as e:
+        raise TypeError('Choicelist elements do not have a common dtype: {}'
+                        .format(e))
```

Comment:
can you use exception chaining here and further down? i.e. `raise TypeError(...) from e`

we get a lot of PRs to add exception chaining (`from e` or `from None`) so we may as well do that whenever we add new code.

Reply:
Right, it should have a `from None` if we include the old message in the new error.  Using an `f` string is a bit nicer, i.e. `f"...: {e}"`.

We don't do this type of replacement a lot, but it does make things clearer, so happy to put this in.